app/code/Bulbulatory/Recommendations/Controller/Adminhtml/Table/Delete.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Controller\Adminhtml\Table;
+
+use Magento\Backend\App\Action;
+use Bulbulatory\Recommendations\Model\RecommendationRepository;
+
+/**
+ * Class Delete
+ * @package Bulbulatory\Recommendations\Controller\Adminhtml\Table
+ */
+class Delete extends Action
+{
+    /**
+     * @var RecommendationRepository
+     */
+    protected $recommendationRepository;
+
+    /**
+     * Delete constructor.
+     * @param Action\Context $context
+     * @param RecommendationRepository $recommendationRepository
+     */
+    public function __construct(
+        Action\Context $context,
+        RecommendationRepository $recommendationRepository
+    )
+    {
+        $this->recommendationRepository = $recommendationRepository;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * @return mixed
+     */
+    protected function _isAllowed()
+    {
+        return $this->_authorization->isAllowed('Bulbulatory_Recommendations::bulbulatory_recommendations_table_delete');
+    }
+
+    /**
+     * @return mixed
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+
+        if ($id = $this->getRequest()->getParam('id'))
+        {
+            try {
+                $recommendation = $this->recommendationRepository->getById($id);
+                $recommendation->delete();

```

Comment:
You created a method in the repository class that deletes the recommendations, it's better to use repository to perform actions in the database than to operate directly on the model object.
```suggestion
                $this->recommendationRepository->delete($recommendation);

```

Reply:
@TM18

> You created a method in the repository class that deletes the recommendations, it's better to use repository to perform actions in the database than to operate directly on the model object.

I changed!